src/utils/temporal.hpp

```diff
@@ -294,6 +293,132 @@ struct LocalDateTimeHash {
   size_t operator()(const LocalDateTime &local_date_time) const;
 };
 
+class Timezone {
+ private:
+  std::variant<std::chrono::minutes, const std::chrono::time_zone *> offset_;
+
+ public:
+  explicit Timezone(const std::chrono::minutes offset) : offset_{offset} {}
+  explicit Timezone(const std::chrono::time_zone *timezone) : offset_{timezone} {}
+  explicit Timezone(std::string_view timezone_name) : offset_{std::chrono::locate_zone(timezone_name)} {}
+
+  const Timezone *operator->() const { return this; }
```

Comment:
This implicit conversion is bad practice. Just use `&thing` at the relevant conversion points.

Reply:
Normally it would be a bad practice, yes. The reason why I added the implicit conversion is because it turned out necessary to make `Timezone` compatible with the interface used by `std::chrono::time_zone`.

See the documentation (_Custom time zone_) in https://howardhinnant.github.io/date/tz.html#Examples:

> One can even have OffsetZone serve as its own smart pointer by giving it a member operator->() that returns itself:
> `const OffsetZone* operator->() const {return this;}`
> This allows you to embed the OffsetZone directly into the zoned_time instead of pointing to an externally held OffsetZone:
> `zoned_time<milliseconds, OffsetZone> zt{OffsetZone{3h + 45min}, floor<milliseconds>(system_clock::now())};`

NB: This behavior is identical in Hinnant’s date library and std::chrono.